tests/transform/test_dynamodb.py

```diff
@@ -84,6 +84,35 @@
     },
     "eventSource": "aws:dynamodb",
 }
+MSG_MODIFY_NESTED = {
+    "awsRegion": "us-east-1",
+    "eventID": "24757579-ebfd-480a-956d-a1287d2ef707",
+    "eventName": "MODIFY",
+    "userIdentity": None,
+    "recordFormat": "application/json",
+    "tableName": "foo",
+    "dynamodb": {
+        "ApproximateCreationDateTime": 1720742302233719,
```

Comment:
We need a test case for the situation of a first level field being an array of objects, for instance
```json
"testarrayobj": [{"field1":1},{"field2":"abc"}]
```
This cast is successful in CrateDB:
```sql
'[{"field1":1},{"field2":"abc"}]'::ARRAY(OBJECT)
```

Reply:
Another situation is the one of a DynamoDB `String Set` for which CrateDB may have already got an `ARRAY(TEXT)` defined, but the actual payload may have a single string, for the `INSERT` or `UPDATE` to succed we need to wrap this in `[ ]`

Reply:
Thank you very much. I am adding your reports to https://github.com/crate/cratedb-toolkit/issues/233#issuecomment-2298485376, in order to merge this patch.